plugins/org.jkiss.dbeaver.ext.intersystems/src/org/jkiss/dbeaver/ext/intersystems/model/InterSystemsMetaModel.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * DBeaver - Universal Database Manager
+ * Copyright (C) 2010-2024 DBeaver Corp and others
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jkiss.dbeaver.ext.intersystems.model;
+
+import java.util.Map;
+
+import org.jkiss.code.NotNull;
+import org.jkiss.dbeaver.DBException;
+import org.jkiss.dbeaver.Log;
+import org.jkiss.dbeaver.ext.generic.model.GenericView;
+import org.jkiss.dbeaver.ext.generic.model.meta.GenericMetaModel;
+import org.jkiss.dbeaver.model.DBPEvaluationContext;
+import org.jkiss.dbeaver.model.DBUtils;
+import org.jkiss.dbeaver.model.exec.jdbc.JDBCPreparedStatement;
+import org.jkiss.dbeaver.model.exec.jdbc.JDBCResultSet;
+import org.jkiss.dbeaver.model.exec.jdbc.JDBCSession;
+import org.jkiss.dbeaver.model.runtime.DBRProgressMonitor;
+import org.jkiss.dbeaver.model.sql.format.SQLFormatUtils;
+
+public class InterSystemsMetaModel extends GenericMetaModel
+{
+    private static final Log log = Log.getLog(InterSystemsMetaModel.class);
+
+    public InterSystemsMetaModel() {
+        super();
+    }
+
+    public String getViewDDL(
+            @NotNull DBRProgressMonitor monitor,
+            @NotNull GenericView view,
+            @NotNull Map<String, Object> options) throws DBException {
+        String sqlStatement = "SELECT View_Definition "
```

Comment:
Does the Intersystem also have the table definition? And procedure? To create more methods about definitions reading. (getTableDDL, getProcedureDDL)

Reply:
Yes, it's possible to get procedure definition, list of triggers, source code for triggers. Do you think it makes sense to do this as part of this pull request? If yes -- I'll do.

Reply:
It can be just definitions. Triggers are also welcome, but you can do them in a separate PR if you want.

Reply:
I added getProcedureDDL and support for listing triggers and getting theirs code. There is no [simple] way to get the table DDL